src/ripple/overlay/impl/PeerSet.cpp

```diff
@@ -84,65 +105,112 @@ PeerSet::addPeers(
 }
 
 void
-PeerSet::setTimer()
+PeerSetImpl::sendRequest(
+    ::google::protobuf::Message const& message,
+    protocol::MessageType type,
+    std::shared_ptr<Peer> const& peer)
 {
-    mTimer.expires_after(mTimerInterval);
-    mTimer.async_wait(
-        [wptr = pmDowncast()](boost::system::error_code const& ec) {
-            if (ec == boost::asio::error::operation_aborted)
-                return;
-
-            if (auto ptr = wptr.lock())
-                ptr->queueJob();
-        });
+    auto packet = std::make_shared<Message>(message, type);
+    if (peer)
+    {
+        peer->send(packet);
+        return;
+    }
+
+    ScopedLockType sl(mLock);
+    for (auto id : mPeers)
+    {
+        if (auto p = app_.overlay().findPeerByShortID(id))
+            p->send(packet);
+    }
 }
 
 void
-PeerSet::invokeOnTimer()
+PeerSetImpl::visitAddedPeers(std::function<void(Peer::id_t)> onPeer)
 {
     ScopedLockType sl(mLock);
+    for (auto id : mPeers)
+    {
+        onPeer(id);
+    }
+}
 
-    if (isDone())
-        return;
+std::size_t
+PeerSetImpl::countAddedPeers()
+{
+    ScopedLockType sl(mLock);
+    return mPeers.size();
+}
```

Comment:
If we make `PeerSet` a container, this should be called `size()`.

Reply:
Now the privately-held container is exposed as a const&. Please see the comment above.